plugins/eos/gs-plugin-eos.c

```diff
@@ -2290,6 +2295,35 @@ gs_plugin_app_upgrade_download (GsPlugin *plugin,
 	return TRUE;
 }
 
+gboolean
+gs_plugin_file_to_app (GsPlugin *plugin,
+		       GsAppList *list,
+		       GFile *file,
+		       GCancellable *cancellable,
+		       GError **error)
+{
+	g_autofree gchar *content_type = NULL;
+	const gchar *mimetypes_repo[] = {
+		"inode/directory",
+		NULL };
+
+	/* does this match any of the mimetypes we support */
+	content_type = gs_utils_get_content_type (file, cancellable, error);
+	if (content_type == NULL)
+		return FALSE;
+	if (g_strv_contains (mimetypes_repo, content_type)) {
```

Comment:
Nitpick: This seems like a fairly roundabout way of checking if `file` is a directory. You could use `g_file_query_file_type()` instead.

Reply:
This matches how the flatpak plugin does this, and how other plugins implement this vfunc